src/compiler/sourcemap.ts

```diff
@@ -0,0 +1,416 @@
+/// <reference path="checker.ts"/>
+
+/* @internal */
+namespace ts {
+    export interface SourceMapWriter {
+        getSourceMapData(): SourceMapData;
+        setSourceFile(sourceFile: SourceFile): void;
+        emitPos(pos: number): void;
+        emitStart(range: TextRange): void;
+        emitEnd(range: TextRange): void;
+        pushScope(scopeDeclaration: Node, scopeName?: string): void;
+        popScope(): void;
+        getText(): string;
+        getSourceMappingURL(): string;
+        initialize(filePath: string, sourceMapFilePath: string, sourceFiles: SourceFile[], isBundledEmit: boolean): void;
+        reset(): void;
+    }
+
+    const nop = <(...args: any[]) => any>Function.prototype;
+    let nullSourceMapWriter: SourceMapWriter;
+
+    export function getNullSourceMapWriter(): SourceMapWriter {
+        if (nullSourceMapWriter === undefined) {
+            nullSourceMapWriter = {
+                getSourceMapData: nop,

```

Comment:
I can add stubs for these, though when I ran the performance tests the cost looked negligible (and the performance tests don't include source maps, so they would be hitting this case). I do wonder if v8 has any optimizations for the Function.prototype primordial. If I see a major difference I'll push up a new commit.